packages/sessions-sdk-ts/src/adapter.ts

```diff
@@ -186,12 +188,16 @@ const buildTransaction = async (
 
 const getSponsor = async (
   options: Parameters<typeof createSolanaWalletAdapter>[0],
+  domain: string,
 ) => {
   if ("sponsor" in options) {
     return options.sponsor;
   } else {
     const response = await fetch(
-      new URL("/api/sponsor_pubkey", options.paymaster ?? DEFAULT_PAYMASTER),
+      new URL(
+        `/api/sponsor_pubkey?domain=${domain}`,
```

Comment:
You should set the domain query param using `searchParams.set` instead so you don't have subtle issues caused by url escaping:

```ts
const url = new URL("/api/sponsor_pubkey", options.paymaster ?? DEFAULT_PAYMASTER);
url.searchParams.set("domain", domain);
const response = await fetch(url);
```

Reply:
done